systemtest/estest/search.go

```diff
@@ -61,10 +61,13 @@ func (es *Client) ExpectMinDocs(t testing.TB, min int, index string, query inter
 		Index:           strings.Split(",", index),
 		ExpandWildcards: "all",
 	}
-	if _, err := refreshReq.Do(context.Background(), es.Transport); err != nil {
+	rsp, err := refreshReq.Do(context.Background(), es.Transport)
```

Comment:
Thank you for the feedback! 

AFAIK the garbage collector does not close the body, that is the caller responsibility.

Closing the body is needed to signal the server that the connection should be closed. Without it, the server keeps idle connections.  
The other advantage is for the client to reuse connections but that is a more obscure behaviour with some runtime magic mixed in (see https://github.com/golang/go/issues/22954).

To observe the behaviour in more detail, consider this:

```
package main

import (
	"fmt"
	"log"
	"net"
	"net/http"
	"net/http/httptrace"
	"time"
)

func main() {
	s := &http.Server{
		Addr: ":8080",
		ConnState: func(_ net.Conn, cs http.ConnState) {
			fmt.Printf("Server got ConnState event: %s\n", cs)
		},
	}
	go func() {
		if err := s.ListenAndServe(); err != nil {
			log.Fatal(err)
		}
	}()

	req, _ := http.NewRequest(http.MethodGet, "http://localhost:8080", nil)

	trace := &httptrace.ClientTrace{
		GotConn: func(i httptrace.GotConnInfo) {
			fmt.Printf("Client got connection: Reused(%v) WasIdle(%v)\n", i.Reused, i.WasIdle)
		},
	}
	req = req.WithContext(httptrace.WithClientTrace(req.Context(), trace))

	send(req)
	send(req)

	time.Sleep(2 * time.Second)
}

func send(req *http.Request) {
	// Example 1
	// http.DefaultClient.Do(req)

	// Example 2
	// http.DefaultClient.Do(req)
	// runtime.GC()

	// Example 3
	// res, _ := http.DefaultClient.Do(req)
	// res.Body.Close()

	// Example 4
	// res, _ := http.DefaultClient.Do(req)
	// io.Copy(io.Discard, res.Body)
	// res.Body.Close()
}
```

Uncomment the examples to see the change in behaviour (Note: client and server are on the different goroutines so the log order is not stable):
____
Example 1 (server is keeping idle connections):

```
Client got connection: Reused(false) WasIdle(false)
Server got ConnState event: new
Server got ConnState event: active
Server got ConnState event: idle
Client got connection: Reused(false) WasIdle(false)
Server got ConnState event: new
Server got ConnState event: active
Server got ConnState event: idle
```

____
Example 2 (same as above, GC didn't help):

```
Server got ConnState event: new
Client got connection: Reused(false) WasIdle(false)
Server got ConnState event: active
Server got ConnState event: idle
Server got ConnState event: new
Client got connection: Reused(false) WasIdle(false)
Server got ConnState event: active
Server got ConnState event: idle
```

____
Example 3 (closing the body closed the connection correctly):

```
Client got connection: Reused(false) WasIdle(false)
Server got ConnState event: new
Server got ConnState event: active
Server got ConnState event: idle
Server got ConnState event: closed
Server got ConnState event: new
Client got connection: Reused(false) WasIdle(false)
Server got ConnState event: active
Server got ConnState event: idle
Server got ConnState event: closed
```

____
Example 4 (close+read allowed the client to reuse the connection):

```
Client got connection: Reused(false) WasIdle(false)
Server got ConnState event: new
Server got ConnState event: active
Server got ConnState event: idle
Client got connection: Reused(true) WasIdle(true)
Server got ConnState event: active
Server got ConnState event: idle
```

Reply:
That's interesting, thanks for providing the examples. I wasn't fully aware of these details when it comes to connection reuse.

Would you say that this change is more about connection reuse than memory reuse, since the body is assigned to a blank variable, thus unused and will be garbage collected, right?